packages/camera/camera_android_camerax/lib/src/camera_control.dart

```diff
@@ -158,11 +171,20 @@ class _CameraControlHostApiImpl extends CameraControlHostApi {
       CameraControl instance, int index) async {
     final int identifier = instanceManager.getIdentifier(instance)!;
     try {
-      return setExposureCompensationIndex(identifier, index);
+      final int? exposureCompensationIndex =
+          await setExposureCompensationIndex(identifier, index);
+      if (exposureCompensationIndex == null) {
```

Comment:
Part of [previously mentioned fix](https://github.com/flutter/packages/pull/6059/files#r1481836864).